glean-core/android/src/test/java/mozilla/telemetry/glean/DispatchersTest.kt

```diff
@@ -60,4 +63,50 @@ class DispatchersTest {
         assertEquals("Tasks have executed", 3, threadCanary)
         assertEquals("Task queue is cleared", 0, Dispatchers.API.taskQueue.size)
     }
+
+    @Test
+    fun `queued tasks are executed in the order they are received`() {
+        var orderedString = ""
+        var curNum = 0
+
+        // Set testing mode to false as we want every call to `launch` to execute as a coroutine
+        Dispatchers.API.setTestingMode(false)
+        Dispatchers.API.setTaskQueueing(true)
+
+        // Spawn a coroutine that will add elements to the orderedString.  This will continue to add
```

Comment:
Why a string? Naturally I would have just reached for a list.

Reply:
> Did you check that the test really (reliably) fails without the changes to `Dispatchers.kt`?

I could get the new test to fail intermittently, and (locally) it appeared that the changes removed the failure.  Of course, it's still failing on CI so I may have to dig a little deeper.

Reply:
A list should work fine. I went for a string as it was easy to compare the two side by side to see when/where it went wrong, but it looks like a list gives me the same ability.